build.sc

```diff
@@ -0,0 +1,122 @@
+import mill._
+import scalalib._
+import mill.scalalib.publish._
+import scala.util.Try
+
+object lsp extends MavenModule with PublishModule {
+
+  def millSourcePath: os.Path = os.pwd
+
+  def ivyDeps = Agg(
+    ivy"org.eclipse.lsp4j:org.eclipse.lsp4j:0.9.0",
+    ivy"software.amazon.smithy:smithy-model:1.14.1",
+    ivy"io.get-coursier:interface:1.0.4"
+  )
+
+  def publishVersion = T { gitVersion() }
+
+  override def artifactName = s"smithy-language-server"
+
+  def pomSettings = PomSettings(
+    description = "LSP implementation for smithy",
+    organization = "com.disneystreaming.smithy",
+    url = "https://github.com/disneystreaming/smithy-language-server",
+    licenses = Seq(License.`Apache-2.0`),
+    versionControl = VersionControl(
+      Some("https://github.com/disneystreaming/smithy-language-server")
+    ),
+    Seq(Developer("baccata", "Olivier Mélois", "https://github.com/baccata"))
+  )
+
+  def gitVersion: T[String] = T.input {
+
+    val gitDirty =
+      os.proc("git", "diff", "HEAD").call().out.lines.nonEmpty
+
+    if (gitDirty) sys.error("Dirty workspace !")
+
+    val commitHash =
+      os.proc("git", "rev-parse", "--short", "HEAD").call().out.lines.head.trim
+
+    val describeResult = os
+      .proc(
+        "git",
+        "submodule",
+        "foreach",
+        "git",
+        "describe",
+        "--long",
+        "--tags",
+        "--abbrev=8",
+        "--match",
+        "v[0-9]*",
+        "--always",
+        "--dirty=+dirty"
+      )
+      .call()
+      .out
+      .lines
+      .lastOption
+      .map(_.replaceAll("-([0-9]+)-g([0-9a-f]{8})", "+$1-$2"))
+      .getOrElse(s"v0.0.0-$commitHash")
+    parseVersion(describeResult)
+  }
+
+  def mainBranch: T[String] = "dss"
+
+  def latestTag: T[Option[String]] = T {
+    val branch = mainBranch()
+    os.proc("git", "describe", branch, "--abbrev=0", "--tags")
+      .call()
+      .out
+      .lines
+      .headOption
+  }
+
+  object int {
+    def unapply(s: String) = Try(s.toInt).toOption
```

Comment:
Nit: I thought Mill is already 2.13 so we can use `toIntOption`

Reply:
src/main/java/software/amazon/smithy/lsp/SmithyProject.java

Reply:
8a04add